packages/view-types/heatmap/src/HeatmapSubscriber.js

```diff
@@ -176,8 +180,14 @@ export function HeatmapSubscriber(props) {
   }, []);
 
   const onHeatmapClick = () => {
-    setGeneSelection([geneHighlight]);
-    setCellColorEncoding('geneSelection');
+    if (localColorEncoding === 'geneSelection') {
+      setGeneSelection([geneHighlight]);
+      setCellColorEncoding('geneSelection');
+    } else if (localColorEncoding === 'cellSelection') {
+      const selectionToHighlight = getObsMembership(cellHighlight);
```

Comment:
I could use `useGetObsInfo` for this too (extract the path from `obsMembership.get(cellInfo[[${capitalize(obsType)} ID]])`). I didn't do it and decided to write a separate function instead, because I didn't want to hardcode `${capitalize(obsType)} ID` in the HeatmapSubscriber, because in the future if we changed the way this is named in `useGetObsMembership` function in hooks, would break this functionality without realising.